orchagent/fgnhgorch.h

```diff
@@ -86,23 +94,30 @@ typedef struct
 class FgNhgOrch : public Orch, public Observer
 {
 public:
-    FgNhgPrefixes fgNhgPrefixes;
     FgNhgOrch(DBConnector *db, DBConnector *appDb, DBConnector *stateDb, vector<string> &tableNames, NeighOrch *neighOrch, IntfsOrch *intfsOrch, VRFOrch *vrfOrch);
 
     void update(SubjectType type, void *cntx);
-    bool addRoute(sai_object_id_t, const IpPrefix&, const NextHopGroupKey&);
-    bool removeRoute(sai_object_id_t, const IpPrefix&);
+    bool isRouteFineGrainedECMP(sai_object_id_t vrf_id, const IpPrefix &ipPrefix, const NextHopGroupKey &nextHops);
+    bool containsRoute(sai_object_id_t vrf_id, const IpPrefix &ipPrefix);
     bool validNextHopInNextHopGroup(const NextHopKey&);
     bool invalidNextHopInNextHopGroup(const NextHopKey&);
+    bool addModifyFgNhg(sai_object_id_t vrf_id, const IpPrefix &ipPrefix, const NextHopGroupKey &nextHops, sai_object_id_t &next_hop_id, bool &prevNhgWasFineGrained);
```

Comment:
>addModifyFgNhg [](start = 9, length = 14)

Do you mean add or modify?
If yes, maybe `setFgNhg` is better.

Reply:
Yes it is add or modify, the fn will either create a new Fine Grained ECMP nexthop group or modify an existing Fine Grained ECMP nexthop group in sai based on some conditions.